### DISCUSSION TITLE
php.CRITICAL

### DISCUSSION THREAD

**Discussion by BeckeBauer** [kimai/kimai#3598] [General]
### Describe the issue

until, yesterday, everything worked well; today, my colleague tried to login and the following error appears:
`php.CRITICAL: Uncaught Error: Call to a member function getId() on null {"exception":"[object] (Error(code: 0): Call to a member function getId() on null at /kimai2/src/Repository/Loader/TimesheetIdLoader.php:52)"} []`
I am still able to login so it seems that just one person is affected... any idea what went wrong?

### I already tried

- [X] I've read and searched [the documentation](https://www.kimai.org/documentation/).
- [X] I've searched for similar issues in this repository.
- [X] I've searched for similar issues in [the discussions](https://github.com/kevinpapst/kimai2/discussions).

### Kimai version

1.27.0

### Which PHP version are you using?

8.0

### How do you run Kimai?

Self-Hosted

### Steps to reproduce the behavior

2. Scroll down to '...'
3. Click on '...'

### Additional context

Device: [Ubuntu Laptop 16 inch, Windows Desktop 27 inch, iPhone 6s]
Browser [e.g. Firefox 81, Chrome 85, Safari 14]

**Comment by kevinpapst** [comment]
That error says that a timesheet without project is existing:
https://github.com/kevinpapst/kimai2/blob/master/src/Repository/Loader/TimesheetIdLoader.php#L52

That is NOT possible in Kimai core, unless someone seriously fiddled around in the database and/or core.
Check the database table "SELECT * FROM kimai2_timesheets where project_id IS NULL"

**Comment by BeckeBauer** [comment]
Strange, no project with ID = NULL found in the database

**Comment by kevinpapst** [comment]
deactivate all plugins (e.g. rename var/plugins/ to var/plugins2/ and mkdir var/plugins/ ) and reload cache. then retry.

**Comment by BeckeBauer** [comment]
got a lot of messages during cache rebuilding, some are

```
Reloading configurations ...
============================

[OK] All 40 YAML files contain valid syntax.                                   
                                                                     
[OK] All 516 XLIFF files contain valid syntax.

Rebuilding your cache, please be patient ...

 // Clearing the cache for the prod environment with debug                      
 // false                                                                       

10:53:37 WARNING   [php] Warning: include(/kimai2/vendor/composer/../../var/plugins/AuditTrailBundle/Doctrine/MetadataSubscriber.php): Failed to open stream: No such file or directory ["exception" => ErrorException { …}]
10:53:37 WARNING   [php] Warning: include(): Failed opening '/kimai2/vendor/composer/../../var/plugins/AuditTrailBundle/Doctrine/MetadataSubscriber.php' for inclusion (include_path='.:/usr/share/pear') ["exception" => ErrorException { …}]
10:53:37 CRITICAL  [php] Uncaught Error: Class "KimaiPlugin\AuditTrailBundle\Doctrine\MetadataSubscriber" not found ["exception" => Error { …}]

In srcApp_KernelProdContainer.php line 1368:
                                                                               
  Attempted to load class "MetadataSubscriber" from namespace "KimaiPlugin\AuditTrailBundle\Doctrine".                                                    
  Did you forget a "use" statement for another namespace?
```
Should I reinstall the plugins?

**Comment by kevinpapst** [comment]
no, you should uninstall them. i want to see if the error is coming from core. do the cache clear once again...

**Comment by BeckeBauer** [comment]
Kimai is working again (without plugins) - colleague is still unable to login:

```
[2022-10-25 11:03:13] php.CRITICAL: Uncaught Error: Call to a member function getCustomer() on null {"exception":"[object] (Error(code: 0): Call to a member function getCustomer() on null at /kimai2/src/Repository/TimesheetRepository.php:617)"} []
[2022-10-25 11:03:13] request.CRITICAL: Uncaught PHP Exception Error: "Call to a member function getCustomer() on null" at /kimai2/src/Repository/TimesheetRepository.php line 617 {"exception":"[object] (Error(code: 0): Call to a member function getCustomer() on null at /kimai2/src/Repository/TimesheetRepository.php:617)"} []
```

**Comment by kevinpapst** [comment]
Same error, different line. There is some data inconsistency in your database.
A timesheet with a project that does not exist or a project with a customer that does not exist.
Something really weird.
You have to check the timesheets of that user. Can you see all of them in Admin > Timesheets?

**Comment by BeckeBauer** [comment]
seems to be the problem ... I am not able to open Admin > Timesheets even though being the system admin... will check the database for weird entries ...

**Comment by kevinpapst** [comment]
Please write down the exact reason! And then we can hopefully find the cause in the AuditTrail Logs.

**Comment by BeckeBauer** [comment]
indeed, found the timesheet entry with a non-existing project number... I will ask my colleague if he has deleted the project after he created the timesheet entry... very strange that this was possible without effecting the timesheet entry...

**Comment by kevinpapst** [comment]
Shouldn't be possible, indeed!
You can check the audit trail logs if there was something removed.

Test if you can reproduce the error. If so: you have to check if all required foreign keys and "on delete" actions are correct on your database (I am afraid you can only compare the structure against a brand new installation). 

worst case option: dump the entire data and import it into a fresh installed system.

**Comment by BeckeBauer** [comment]
Thanks for the help - I will take care of it when I return from vacations (such things happens of course always during absence...)

**Reply by kevinpapst** [reply]
Of course 😁 enjoy your holiday!
